src/modules/inventory/src/inventoryImp.cpp

```diff
@@ -618,87 +595,63 @@ nlohmann::json Inventory::GetNetworkData()
+
+                        ret[NETWORKS_TABLE].push_back(networkTableData);
+
                     }
                 }
 
                 if (item.find("IPv6") != item.end())
                 {
-                    // "dbsync_network_protocol" table data to update and notify
-                    nlohmann::json protoTableData {};
-                    protoTableData["iface"]   = item.at("name");
-                    protoTableData["gateway"] = item.at("gateway");
-                    protoTableData["type"]    = IP_TYPE.at(IPV6);
-                    protoTableData["dhcp"]    = item.at("IPv6").begin()->at("dhcp");
-                    protoTableData["metric"]  = item.at("IPv6").begin()->at("metric");
-                    protoTableData["checksum"]  = GetItemChecksum(protoTableData);
-                    protoTableData["item_id"]   = GetItemId(protoTableData, NETPROTO_ITEM_ID_FIELDS);
-                    protoTableDataList.push_back(std::move(protoTableData));
-
                     for (auto addressTableData : item.at("IPv6"))
                     {
-                        // "dbsync_network_address" table data to update and notify
-                        addressTableData["iface"]     = item.at("name");
-                        addressTableData["proto"]     = IPV6;
-                        addressTableData["checksum"]  = GetItemChecksum(addressTableData);
-                        addressTableData["item_id"]   = GetItemId(addressTableData, NETADDRESS_ITEM_ID_FIELDS);
-                        // Remove unwanted fields for dbsync_network_address table
-                        addressTableData.erase("dhcp");
-                        addressTableData.erase("metric");
-
-                        addressTableDataList.push_back(std::move(addressTableData));
+                        nlohmann::json networkAddressData {};
+                        networkAddressData["proto_type"]  = IP_TYPE.at(IPV6);
+                        networkAddressData["address"] = addressTableData.at("address");
+                        networkAddressData["broadcast"] = addressTableData.at("broadcast");
+                        networkAddressData["dhcp"]    = addressTableData.at("dhcp");
+                        networkAddressData["metric"]  = addressTableData.at("metric");
+                        networkAddressData["netmask"] = addressTableData.at("netmask");
+                        networkTableData.update(networkAddressData);
+
+                        networkTableData["network_checksum"]   = GetItemChecksum(networkTableData);
+                        networkTableData["network_item_id"]    = GetItemId(networkTableData, NETWORK_ITEM_ID_FIELDS);
+
+                        ret[NETWORKS_TABLE].push_back(networkTableData);
                     }
                 }
             }
-
-            ret[NET_IFACE_TABLE] = std::move(ifaceTableDataList);
-            ret[NET_PROTOCOL_TABLE] = std::move(protoTableDataList);
```

Comment:
Is it possible to have an unaddressed interface?

Reply:
AFAIK there's no chance of getting an interface without an address, in those cases the data provider is throwing an exception. Example on sysInfoLinux.cpp:

```cpp
        nlohmann::json ifaddr {};

        for (auto addr : interface.second)
        {
            const auto networkInterfacePtr { FactoryNetworkFamilyCreator<OSPlatformType::LINUX>::create(std::make_shared<NetworkLinuxInterface>(addr)) };

            if (networkInterfacePtr)
            {
                networkInterfacePtr->buildNetworkData(ifaddr);
            }
        }

        networks["iface"].push_back(ifaddr);
```

```cpp
void LinuxNetworkImpl<AF_INET>::buildNetworkData(nlohmann::json& network)
{
    // Get IPv4 address
    const auto address { m_interfaceAddress->address() };

    if (!address.empty())
    {
        nlohmann::json ipv4JS { };
        ipv4JS["address"] = address;
        ipv4JS["netmask"] = m_interfaceAddress->netmask();
        ipv4JS["broadcast"] = m_interfaceAddress->broadcast();
        ipv4JS["metric"] = m_interfaceAddress->metrics();
        ipv4JS["dhcp"]   = m_interfaceAddress->dhcp();

        network["IPv4"].push_back(ipv4JS);
    }
    else
    {
        throw std::runtime_error { "Invalid IpV4 address." };
    }
}
```